Xamarin.Forms.Core/Element.cs

```diff
@@ -343,11 +343,14 @@ protected override void OnBindingContextChanged()
 
 		protected virtual void OnChildAdded(Element child)
 		{
+			var bindingContext = child.BindingContext;
+
 			child.Parent = this;
 			if (Platform != null)
 				child.Platform = Platform;
 
-			child.ApplyBindings();
+			if(bindingContext == child.BindingContext)
+				child.ApplyBindings();
```

Comment:
In general, setting `Parent` should set child binding context, but there are two checks in parent set that return before `SetChildInheritedBindingContext()` is called. It seemed safer to add this check instead of removing apply bindings altogether. I'll work on unit testing tomorrow.